python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -344,20 +363,35 @@ def empty(cls) -> "AssetReconciliationCursor":
             materialized_or_requested_root_partitions_by_asset_key={},
             materialized_or_requested_root_asset_keys=set(),
             evaluation_id=0,
+            last_observe_request_timestamp_by_asset_key={},
         )
 
     @classmethod
     def from_serialized(cls, cursor: str, asset_graph: AssetGraph) -> "AssetReconciliationCursor":
         data = json.loads(cursor)
-        check.invariant(len(data) in [3, 4], "Invalid serialized cursor")
 
-        (
-            latest_storage_id,
-            serialized_materialized_or_requested_root_asset_keys,
-            serialized_materialized_or_requested_root_partitions_by_asset_key,
-        ) = data[:3]
+        if isinstance(data, list):  # backcompat
+            check.invariant(len(data) in [3, 4], "Invalid serialized cursor")
+            (
+                latest_storage_id,
+                serialized_materialized_or_requested_root_asset_keys,
+                serialized_materialized_or_requested_root_partitions_by_asset_key,
+            ) = data[:3]
 
-        evaluation_id = data[3] if len(data) == 4 else 0
+            evaluation_id = data[3] if len(data) == 4 else 0
+            serialized_last_observe_request_timestamp_by_asset_key = {}
+        else:
```

Comment:
this is getting pretty nasty -- might be nice to swap over to the `whitelist_for_serdes` serializer

Reply:
agree that this is nasty, but alas I don't think there's a straightforward way to avoid the nastiness: https://github.com/dagster-io/dagster/pull/14149/files#r1255004744